src/psbt.cpp

```diff
@@ -215,6 +215,25 @@ bool PSBTInputSigned(const PSBTInput& input)
     return !input.final_script_sig.empty() || !input.final_script_witness.IsNull();
 }
 
+void UpdatePSBTOutput(const SigningProvider& provider, PartiallySignedTransaction& psbt, int index)
```

Comment:
Could name this `UpdatePSBT()` and iterate outputs here? - which is what both callers do.

Reply:
Alternatively, seems this could be appropriate as `PartiallySignedTransaction::UpdateOutput`